Tools/TestWebKitAPI/Tests/WTF/CompletionHandlerTests.cpp

```diff
@@ -137,6 +138,25 @@ TEST_F(CompletionHandlerTest, ConstructWithSpecificThreadLikeAssertion)
     EXPECT_TRUE(didCall);
 }
 
+TEST_F(CompletionHandlerTest, FinalizerInheritsHandlersThreadAssertion)
+{
+    CompletionHandler<void()> handler;
+    Thread::create("ConstructionThread"_s, [&] {
+        handler = CompletionHandler<void()> { [] { }, CompletionHandlerCallThread::AnyThread };
+    })->waitForCompletion();
+
+    CompletionHandlerWithFinalizer<void()> handlerWithFinalizer { WTFMove(handler), [](auto&& callable) {
+        callable();
+    } };
+
+    bool didDestroy = false;
+    Thread::create("DestructionThread"_s, [&] {
+        auto localHandlerWithFinalizer = WTFMove(handlerWithFinalizer);
+        didDestroy = true;
```

Comment:
Your proof of work in here is inconclusive, as it would pass in correct behaviour, but also if both functions were run, and if no functions were run.

In the commit message your proof of work is ambiguous, as it's not entirely clear if the the function itself is destroyed in the thread or not. It is in current implementation, except in error cases. Naturally in commit message you wanted to keep it short..